src/PRASBase/pras_system_size_reducer.jl

```diff
@@ -0,0 +1,91 @@
+#######################################################
+# Surya
+# NREL
+# January 2020
+# Make a smaller PRAS system (with a fewer timesteps)
+# from a bigger PRAS system
+#######################################################
+# Main Function to reduce size of PRAS System
+#######################################################
+function reduce_pras_system_size(pras_system::PRAS.SystemModel,period_of_interest::UnitRange)
+    """
+    reduce_system_size(sys,x)
+
+    Build a smaller PRAS system with 'x' time_steps from a larger PRAS System.
+    
+    ...
+    # Arguments
+    - `sys::SystemModel`: Original PRAS System.
+    - `x::UnitRange`: Period of interest to build smaller PRAS System
+    ...
+
+    # Examples
+    ```julia-repl
+    julia> reduce_system_size(sys,range(100,length=100))
+    SystemModel{100,timestep_length,timeunits,powerunits,energyunits(optional)}
+    ```
+    """
+    symbol = PRAS.unitsymbol(pras_system);
+    time_steps = length(pras_system.regions.load[1,:]);
+    
+    time_units_symbol = symbol[1];
+    power_units_symbol = symbol[2];
+    energy_units_symbol = symbol[3];
+
+    time_units = PRAS.PRASBase.timeunits[time_units_symbol];
+    power_units = PRAS.PRASBase.powerunits[power_units_symbol];
+    energy_units = PRAS.PRASBase.energyunits[energy_units_symbol];
+    
+    sys_type = typeof(pras_system);
+
+    @info "The original system is of type $(sys_type). You are trying to build a system of type SystemModel{$(length(period_of_interest)),
+           1,$(time_units),$(power_units),$(energy_units)}."
```

Comment:
Following on from the previous comment, you can use `L,T,P,E` here. Also, since `length(period_of_interest)` is repeated multiple times it might be cleaner just to precompute it as `new_N = length(period_of_interest)`. Then this becomes:

```julia
"... of type SystemModel{$(new_N),$L,$T,$P,$E}."
```

Reply:
Yeah, precomputing new_N makes sense. Included in my latest commit.